TradeItIosTicketSDK2/TradeItTransactionsTableViewManager.swift

```diff
@@ -17,17 +17,30 @@ class TradeItTransactionsTableViewManager: NSObject, UITableViewDelegate, UITabl
     }
     private static let HEADER_HEIGHT = CGFloat(30)
     private static let CELL_HEIGHT = CGFloat(65)
-    private var transactionHistoryResultPresenter: TransactionHistoryResultPresenter?
+    var transactionHistoryResultPresenter: TransactionHistoryResultPresenter?
     private var linkedBrokerAccount: TradeItLinkedBrokerAccount
-    
+    private var noResultsBackgroundView: UIView
+    private var filterType: TransactionFilterType = TransactionFilterType.ALL_TRANSACTIONS
     weak var delegate: TradeItTransactionsTableDelegate?
     
-    init(linkedBrokerAccount: TradeItLinkedBrokerAccount) {
+    init(linkedBrokerAccount: TradeItLinkedBrokerAccount, noResultsBackgroundView: UIView) {
         self.linkedBrokerAccount = linkedBrokerAccount
+        self.noResultsBackgroundView = noResultsBackgroundView
     }
     
     func updateTransactionHistoryResult(_ transactionHistoryResult: TradeItTransactionsHistoryResult) {
-        self.transactionHistoryResultPresenter = TransactionHistoryResultPresenter(transactionHistoryResult, accountBaseCurrency: self.linkedBrokerAccount.accountBaseCurrency)
+        self.transactionHistoryResultPresenter = TransactionHistoryResultPresenter(transactionHistoryResult, filterType: filterType, accountBaseCurrency: self.linkedBrokerAccount.accountBaseCurrency)
+        if let transactions = transactionHistoryResult.transactionHistoryDetailsList, !transactions.isEmpty {
+            self.transactionsTable?.backgroundView =  nil
+        } else {
+            self.transactionsTable?.backgroundView = noResultsBackgroundView
+        }
+        self.transactionsTable?.reloadData()
+    }
+
+    func filterTransactionHistoryResult(filterType: TransactionFilterType) {
+        self.filterType = filterType
```

Comment:
Where does this get used? The `filterType` is also stored on the presenter. Would be nice to have only one place to keep track of this.